### PR TITLE
Change Statix functions to be overridable

### PR DESCRIPTION
**Pull request by edgurgel**
First thing: thanks for this awesome lib :)

I found myself wanting to override the functions so that for example I can conditionally add some tags, change the key name etc. Would this be useful for others? 

I'm also not sure if the best solution is actually to make `transmit` part of the "main module" and make just that overridable?

And the tests are pretty much a copy and paste for now.

Thanks in advance :)

### CODE REVIEW

File: lib/statix.ex [hunk 1]
```diff
@@ -313,6 +313,14 @@ defmodule Statix do
       def set(key, val, options \\ []) do
         Statix.transmit(current_conn(), :set, key, val, options)
       end
+
+      defoverridable [increment: 1, increment: 2, increment: 3,
```
Comment (lexmag) [inline lib/statix.ex]: Maybe we should make only the top-arity functions overridable?

Comment (edgurgel) [reply lib/statix.ex]: Agreed!

Comment (edgurgel) [reply lib/statix.ex]: Done! 👍

**Comment by lexmag** [comment]
Thank you @edgurgel. 💛

**Comment by edgurgel** [comment]
Thank you! <3

**Pull request merged**